lib/services/payment_tracking/payment_tracking_config.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+
+/// Types of payment tracking supported by the application
+enum PaymentTrackingType { bitcoinTransaction, lightningAddress, lightningInvoice, none }
```

Comment:
It seems to me that the abstractions bellow for paymentTrackingConfig and also the PaymentTrackingType and PaymentType enums are only meant for logging purposes.
I would consider reducing it to a very simple common subscription registered by the caller and let the caller log additional info if required.
Also it seems that a match function (serving as filter for the event) can be passed from the caller reducing the needed abstraction.

Unless I am missing anything  seems to me like a simple wrapper over the sdk events system should be enough.